I've updated the changefile to reflect both changed functions, like @chippers suggested in #2290.

I do want to note that this may be considered a breaking change (which is why I marked it as 'major' in the changefile). Especially because @chippers said this isn't a major change and I don't entirely agree. (I was going to comment in the issue but I feel it's more appropriate to continue this conversation here in the PR).

> One thing to be changed is the changefile level, this should be a patch not major change

Because while the type is technically backwards-compatible (since `String` is also `Into<String>`), in practice most developers who use this function will need to update their code.

The current `create_window()` function takes a `String` for the label. So when a developer uses that function to create a window, they'll do something like this:
```rs
app.create_window("my_label".into(), [...]);
```

Note the `.into()`. This is also how it's done in the example.

With this PR, `create_window()` accepts any `Into<String>` type for the label, so developers will be able to just do this:
```rs
app.create_window("my_label", [...]);
```

However, when trying to use the original code with `.into()` after applying this change, you get the following error:
```
error[E0283]: type annotations needed
  --> examples\api\src-tauri\src\main.rs:84:14
   |
84 |             .create_window(
   |              ^^^^^^^^^^^^^ cannot infer type for type parameter `impl Into<String>` declared on the associated function `create_window`
85 |               "new".into(),
   |               ------------ this method call resolves to `T`
   |
   = note: cannot satisfy `_: Into<std::string::String>`
```

So because the `impl Into<String>` type doesn't point to a concrete type, `.into()` can't determine what type to convert the value into. This means that most (if not all) developers who use this function will be presented with these errors when they upgrade their Tauri version and they'll have to fix them to get their app to compile. Theoretically the function still accepts the same type as before and this PR just expands the accepted types a bit, but I would in fact consider this a major change because it *will* break code.

If you don't consider this a breaking change, let me know and I'll bring the changefile down to a patch level.

--- a/.changes/fix-create-window-str.md
+++ b/.changes/fix-create-window-str.md
@@ -1,5 +1,5 @@
 ---
-"tauri": major
+"tauri": patch
 ---
 
 Change `App.create_window()` and `AppHandle.create_window()` to accept an `Into<String>` type instead of `String`.

